### ISSUE TITLE
One unitig is unexpectedly missing after HiC phasing

### ISSUE THREAD

**Issue by zhangrengang** [chhylp123/hifiasm#159]
The below is a unitig graph: the unitigs A and B is allelic (confirmed by aligning to reference). After HiC phasing, unitig A is in hic.hap2.p_ctg, but unitig B is missing in hic.hap1.p_ctg.
![360截图1674010985115114](https://user-images.githubusercontent.com/22489982/126920866-f8a1b38b-4bd9-43e6-8f40-b425ca297e33.png)

The following is the corresponding HiC heatmap:
![360截图16740117171547](https://user-images.githubusercontent.com/22489982/126921058-9ea969c5-1891-4405-9b87-7e7dad27c2b1.png)

By the way, some switch errors are observed, but they can be corrected manually.

**Comment by chhylp123** [comment]
It is possible... how long of A and B?

**Comment by zhangrengang** [comment]
About 1M bp.

**Comment by zhangrengang** [comment]
In total, two issues of this type are observed in my assembly: one is ~1 Mbp (the above) and the other is ~400 Kbp. The 400kb is somewhat different as follows: the unitig C seems to be homozygous; it is in hap1 but not in hap2 after HiC phasing.
![360截图17571118107135135](https://user-images.githubusercontent.com/22489982/126926494-5ee99539-f911-4898-bd52-fcc81b531928.png)
![360截图17401210488969](https://user-images.githubusercontent.com/22489982/126926612-91e0af6f-2447-48a1-8d8b-87362dd18c73.png)

**Comment by chhylp123** [comment]
Thanks. Can you also find unitig B in hap2?

**Comment by zhangrengang** [comment]
No. I combine hap1 and hap2 in HiC heatmap, so it will be present in the heatmap if it is in either haps.

**Comment by zhangrengang** [comment]
No. I combine hap1 and hap2 in HiC heatmap, so it will be present in the heatmap if it is in either haps.

**Comment by chhylp123** [comment]
It is interesting. Is this possible that you can share the bin file with us for debugging?

**Comment by zhangrengang** [comment]
I am sorry that the data is not my own, so I can not share.

**Comment by chhylp123** [comment]
I see. Thanks for reporting this issue.

**Comment by chhylp123** [comment]
Could you please have a try with the current release (https://github.com/chhylp123/hifiasm/releases/tag/0.16.1)? I guess it should be able to resolve this issue.

**Comment by zhangrengang** [comment]
Thanks. I have tried and the missed unitig is present this time.

**Comment by chhylp123** [comment]
Thanks a lot!
